build.yaml

```diff
@@ -196,13 +213,21 @@ common_cli:
     # SSL on our caching box.
     # The --trusted-host option is only available with pip 7.
     #
-    parse_logs pip install --upgrade pip
-    # using the caching-layer, install all the dependencies
-    parse_logs pip install -i \$PIP_INDEX_URL . --trusted-host \$TRUSTED_HOST
-    # using the caching-layer, install flocker
-    parse_logs pip install -i \$PIP_INDEX_URL "Flocker[dev]" --trusted-host \$TRUSTED_HOST
-    # using the caching-layer, install junix for our coverage report
-    parse_logs pip install -i \$PIP_INDEX_URL python-subunit junitxml  --trusted-host \$TRUSTED_HOST
+    if [ "\$(os)" == "ubuntu" ] || [ "\$(os)" == "centos" ]; then
+        parse_logs pip install --upgrade pip
+        # using the caching-layer, install all the dependencies
+        parse_logs pip install -i \$PIP_INDEX_URL . --trusted-host \$TRUSTED_HOST
+        # using the caching-layer, install flocker
+        parse_logs pip install -i \$PIP_INDEX_URL "Flocker[dev]" --trusted-host \$TRUSTED_HOST
+        # using the caching-layer, install junix for our coverage report
+        parse_logs pip install -i \$PIP_INDEX_URL python-subunit junitxml  --trusted-host \$TRUSTED_HOST
+    fi
+    if [ "\$(os)" == "OSX" ]; then
+      pip install --upgrade pip
+      pip install .
+      pip install  "Flocker[dev]"
+      pip install python-subunit junitxml
+    fi
 
```

Comment:
we're not injecting PIP caching env variables to our OSX boxes, that is done on centos and ubuntu through the slave_setup jenkins plugin, which we can't execute on cloudbees